core/services/gcp_service.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'googleauth'
+require 'google/apis/compute_v1'
+
+# This class allows to execute commands in accordance to the Google Cloud Compute
+class GcpService
+  SCOPE = %w[https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/compute]
+
+  def initialize(gcp_config, logger)
+    @gcp_config = gcp_config
+    @logger = logger
+    @service = Google::Apis::ComputeV1::ComputeService.new
+    @service.authorization = Google::Auth::ServiceAccountCredentials.make_creds(
+      json_key_io: File.open(@gcp_config['credentials_file']),
+      scope: SCOPE
+    )
+    @service.authorization.fetch_access_token!
+  end
+
+  # Fetch instances list and return instance names.
+  # @return [Array<String>] instance names.
+  def instances_list
+    items = @service.fetch_all do |token|
```

Comment:
You do not need items variable here. Just chain the `map` method to the results of calling `fetch_all`.